src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,138 @@
+    private static final String VALID_LOGIN1 = "Andrew";
+    private static final String VALID_LOGIN2 = "Penelopa";
+    private static final String VALID_PWD = "1234567";
+    private static final int VALID_AGE = 20;
+    private static final String INVALID_LOGIN = "Andr";
+    private static final String INVALID_PWD = "12345";
+    private static final int INVALID_AGE = 17;
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LENGTH = 6;
+    private static final User INVALID_LOGIN_USER = new User(INVALID_LOGIN, VALID_PWD, VALID_AGE);
+    private static final User INVALID_PWD_USER = new User(VALID_LOGIN1, INVALID_PWD, VALID_AGE);
+    private static final User INVALID_AGE_USER = new User(VALID_LOGIN1, VALID_PWD, INVALID_AGE);
+    private static final User VALID_USER = new User(VALID_LOGIN2, VALID_PWD, VALID_AGE);
+    private static final User VALID_USER2 = new User("Alexandr", "password", 50);
+    private static final User VALID_USER3 = new User("Leopold", "qwerty", 33);
+    private static RegistrationServiceImpl regService;
+
+    @BeforeAll
+    static void beforeAll() {
+        regService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void register_nullUser_Exception_notOk() {
+        RegistrationException invalidDataException = assertThrows(RegistrationException.class,
+                () -> {
+                regService.register(NULL_USER);
+                });
+        assertEquals("User can't be null", invalidDataException.getMessage());
+    }
+
+    @Test
+    void register_nullPassword_Exception_notOk() {
+        RegistrationException invalidDataException = assertThrows(RegistrationException.class,
+                () -> {
+                regService.register(new User(VALID_LOGIN1, NULL_PASSWORD, VALID_AGE));
+                });
+        assertEquals("Password can't be null", invalidDataException.getMessage());
+    }
+
+    @Test
+    void register_nullLogin_Exception_notOk() {
+        RegistrationException invalidDataException = assertThrows(RegistrationException.class,
+                () -> {
+                regService.register(new User(NULL_LOGIN, VALID_PWD, VALID_AGE));
+                });
+        assertEquals("Login can't be null", invalidDataException.getMessage());
+    }
+
+    @Test
+    void register_nullAge_Exception_notOk() {
+        RegistrationException invalidDataException = assertThrows(RegistrationException.class,
+                () -> {
+                regService.register(new User(VALID_LOGIN1, VALID_PWD, NULL_AGE));
+                });
+        assertEquals("Age can't be null", invalidDataException.getMessage());
+    }
+
+    @Test
+    void register_invalidLogin_Exception_notOk() {
+        RegistrationException invalidDataException = assertThrows(RegistrationException.class,
+                () -> {
+                regService.register(INVALID_LOGIN_USER);
+                });
+        assertEquals("Not valid login length: "
```

Comment:
Please add some more cases for login and password lenght check, for example (0, 5, 3). 3 - is regular invalid case, 5 - edge case, 0 - edge case